### ISSUE TITLE
Bug: Auto-Add to Shopping List Creates List Items with "Negative Quantity"

### ISSUE THREAD

**Issue by crossan007** [grocy/grocy#2154]
# Summary

Reproduced on: https://test-hy49t5m3lk6yyinpu6n0a.demo.grocy.info

Given a parent/child product structure of 

```
- Chips (no own stock; minimum quantity 5)
  + Garden Salsa Sun Chips (minimum quantity 1; 3 in stock)
  + Tortilla Chips (minimum quantity 1; 3 in stock)
  + Sour Cream and Onion Chips (no minimum quantity; none in stock)
```

Opening the Shopping list and clicking `Add products that are below defined min. stock amount` causes "Chips" to be added to the shopping list with a quantity of `-1`; as if to say "Get rid of one bag of chips to achieve the minimum quantity"

# Expected Behavior

 Since the minimum quantity of the parent product ("Chips") is met by the 6 total child products; there should be no entry added to the shopping list for "Chips"

# Possibly Related Issues
*  https://github.com/grocy/grocy/issues/1804

**Comment by seeker** [comment]
Is this a regression?

I can reproduce this issue on 3.3.2:
[https://test-kw2atxxiqeubyzr1soiiq.demo.grocy.info/shoppinglist?list=1](https://test-kw2atxxiqeubyzr1soiiq.demo.grocy.info/shoppinglist?list=1)

The item `test_item_parent` has a min. value of 5, the `test_item_child` has the previous item as it's parent and a stock of 13.
When clicking on "Add products that are below defined min. stock amount", `test_item_parent` is added with -8.

**Comment by berrnd** [comment]
> Is this a regression?

Please mind the milestone. This is not yet part of any release, so not a real surprise that anything reported here is still the case there.

**Comment by seeker** [comment]
My apologies, I had assumed this was fixed in 3.3.2, as the commit that closed this issue is part of 3.3.2.
